### PR TITLE
Don't cancel saga when put dispatch throws

### PR DESCRIPTION
**Pull request by michaelgilley**
Ref #550 

When putting onto the store via dispatch the operation should be considered an emit on the Redux store that does not bubble up errors from reducers/helpers/etc. subsequently cancelling the saga.

This patch simply logs such errors and renders the put as a noop.

### CODE REVIEW

**Comment by yelouafi** [comment]
It makes sense as the put effect is non blocking. A couple of remarks.

We could make the handling more consistent. Right now the put effect can be either blocking (using `put.sync(...)`) or non blocking (the default, using `put(...)`). So what I propose is

- If the put is non-blocking (`sync == false`) then we do not bubble the error, just log it as in the PR.

- If the put is blocking (`sync == true`) then keep the actual behavior.

For logging, I noticed that logging `error.stack` gives better error logging in console.chrome. Relying on console.error (invoked by `log('error', ...)`) doesn't show where the error exactly occured (in this example inside the `checkoutStatus` reducer of the shopping cart example)

err-1

logging error.stack explicitly shows the orignial stack of the error

err-2

**Comment by michaelgilley** [comment]
I agree on both accounts. I have updated the PR.

A few notes however:

1. This behavior should be documented
1. `put.sync` should be added to the documentation
1. Logging is [inconsistent](https://github.com/yelouafi/redux-saga/blob/master/src/internal/proc.js#L353) and at times unhelpful (as you pointed out here). I vote we make error reporting more robust.

File: src/internal/proc.js [hunk 1]
```diff
@@ -441,7 +441,7 @@ export default function proc(
         result = (channel ? channel.put : dispatch)(action)
       } catch(error) {
         if (channel) return cb(error, true)
-        log('error', `uncaught at ${name}`, error.message)
+        log('error', `uncaught at ${name}`, error.stack)
```
Comment (yelouafi) [inline src/internal/proc.js]: I think we should check if `error instanceof Error` because dev. could throw other things (like `throw "some error message"`

**Comment by yelouafi** [comment]
>This behavior should be documented

Would you like to add docs as part of this PR ?

>put.sync should be added to the documentation

Yeah! forgot it 😅

>Logging is inconsistent and at times unhelpful (as you pointed out here). I vote we make error reporting more robust.

You're right. Do you have any other suggestions to improve error reporting ? (you can open a separate issue for this)

File: src/internal/proc.js (hunk 1 above)
Comment (michaelgilley) [inline src/internal/proc.js]: Yeah, this is the sort of thing that should be handled by the logger. For now I'll add a check though.

**Comment by michaelgilley** [comment]
I've add error handling for put in the API docs and I'll raise the log reporting in another issue.

**Comment by yelouafi** [comment]
Looks fine. Thanks!

**Pull request merged**